tests/codemods/test_with_threading_lock.py

```diff
@@ -86,3 +86,56 @@ def test_no_effect_multiple_with_clauses(self, tmpdir, klass):
     ...
 """
         self.run_and_assert(tmpdir, input_code, input_code)
+
+
+class TestThreadingNameResolution(BaseSemgrepCodemodTest):
```

Comment:
@andrecsilva wondering if these are the cases you're referring to by suggesting to implement get current names at init? because right now the codemod will change
```
with Lock():
    with Lock():
        foo()
```

to 
```
import threading
lock = threading.Lock()
with lock:
    lock = threading.Lock()
    with lock:
        foo()
```

which is not great since it itself is adding a name clash. This is happening because both calls to `leave_With` happen one after the other, with the second one not "learning about"  the new addition of the first `lock = ...`. I don't believe this would be solved by  moving `current_names = self.find_used_names_in_module()` to codemod init because that too will happen only once.

Reply:
Yes, this was what I was afraid it would happen. 
My suggestion is moving the calculation of `current_names` calculation to `WithThreadingLock`'s `__init__` and update `current_names` with the newly added name after `leave_With`.

Reply:
ahh I see what you mean now.